src/battle_util.c

```diff
@@ -6458,35 +6458,42 @@ static inline bool32 CanBreakThroughAbility(u32 battlerAtk, u32 battlerDef, u32
 
 u32 GetBattlerAbility(u32 battler)
 {
-    bool32 noAbilityShield = GetBattlerHoldEffectIgnoreAbility(battler, TRUE) != HOLD_EFFECT_ABILITY_SHIELD;
-    bool32 abilityCantBeSuppressed = gAbilitiesInfo[gBattleMons[battler].ability].cantBeSuppressed;
-
-    if (abilityCantBeSuppressed)
+    if (gBattleStruct->aiCalcInProgress)
```

Comment:
Actually, I've been thinking do we need this? 

When I removed AI function I added an argument for ability so the right one can be passed depending where it is called from.

Reply:
I checked and as far as I can see the only 2 remaining functions to port are `AI_IsTerrainAffected` and `AI_IsBattlerGrounded`. In that case I would just use an argument for them so the `aiCalcInProgress` wouldn't be needed anymore